### PR TITLE
feat: use `jshttp/mime-db` combined db

### LINKED ISSUE #13: Keynote file type
Apple Keynote files (.key, application/vnd.apple.keynote) isn't listed. Do you not refer to https://www.iana.org/assignments/media-types/media-types.xhtml ?

D

### LINKED ISSUE #15: Apache mime type list seems outdated. Why don't build this library on iana list?
I notice I bring an example: I verified that sqlite is not even mentioned in the apache list. Moreover "recently" iana has changed the standard (https://stackoverflow.com/a/55071463) and has updated its list accordingly.

I understand that iana doesn't provide the list of extensions together with the mime, but at least you could be able to validate the mime type. So at least you could put this list next to the one used with the file extension.

What is your opinion about this?

thanks

### PR DESCRIPTION
**Pull request by viceice**
use combined db from `jshttp/mime-db` instead of multiple others

- closes #13
- closes #15

### CODE REVIEW

File: KnownMimeTypes.tt [hunk 1]
```diff
@@ -16,13 +16,11 @@ using System;
 namespace MimeMapping
 {
 <#
-    const string APACHE_URL = "http://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types";
-    const string NGINX_URL = "https://raw.githubusercontent.com/h5bp/server-configs-nginx/master/mime.types";
-	const string MIMEDB_URL = "https://raw.githubusercontent.com/jshttp/mime-db/master/src/custom-types.json";
+	const string MIMEDB_URL = "https://cdn.jsdelivr.net/gh/jshttp/mime-db@master/db.json";
```
Comment (zone117x) [inline KnownMimeTypes.tt:19]: Looking at the readme for that repo:
> If you're crazy enough to use this in the browser, you can just grab the JSON file using [jsDelivr](https://www.jsdelivr.com/). 

In our case, we're grabbing this data in a source generation script which is rarely ran. In that case, probably makes more sense to grab directly from github as opposed to this cdn.

> It is recommended to replace master with [a release tag](https://github.com/jshttp/mime-db/tags) as the JSON format may change in the future.

It sounds like we should also follow this advice, so this repo doesn't randomly break in the future if the format changes. And a specific tag will make the set of mime types supported by this repo will be more clear.

**Review by zone117x — COMMENTED** [review]
Thanks for the PR @viceice! 

I think this is probably a good direction for this repo. If jshttp/mime-db is already aggregating sources and well maintained, there's not a very compelling reason to duplicate the effort in this repo. 

Could you also update this description string: https://github.com/zone117x/MimeMapping/blob/d020c402dfb542cda16ab9afcef8f522dbd2f698/MimeMapping.csproj#L5-L12

I think the description should also still mention the sources used to compile the data, e.g. could just include a quote from their readme:
> It aggregates data from the following sources:
> https://www.iana.org/assignments/media-types/media-types.xhtml
> https://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types
> https://hg.nginx.org/nginx/raw-file/default/conf/mime.types


Also, the readme for this repo needs a similar update.

File: KnownMimeTypes.tt [hunk 2]
```diff
@@ -63,19 +61,6 @@ namespace MimeMapping
 
         var entries = new List<string[]>();
 
-		entries.AddRange(GetMimeTypesFromText(APACHE_URL, (line) => {
```
Comment (zone117x) [inline KnownMimeTypes.tt:66]: Looks like the `GetMimeTypesFromText` function can also now be safely removed.

**Comment by viceice** [comment]
finally works 🥳  Should probably released as v1.1.0 as it's changing a lot mime types

File: MimeMapping.csproj [hunk 3]
```diff
@@ -4,11 +4,15 @@
     <TargetFrameworks>netstandard1.1;netstandard2.0;netstandard2.1</TargetFrameworks>
     <Description>Constants for (almost) all MIME types and method to determine MIME type from a file name. Contains just over 1000 mime types.
 
-Generated from the Apache server mime.types file and the H5PB nginx mime.types file. Works similar to .NET's System.Web.MimeMapping.GetMimeMapping.
+Generated from the jshttp/mime-db dp.json file. Works similar to .NET's System.Web.MimeMapping.GetMimeMapping.
```
Comment (zone117x) [inline MimeMapping.csproj:7]: Nit: link to https://github.com/jshttp/mime-db ?

Comment (zone117x) [reply MimeMapping.csproj:7]: Nvm, I see below :)

**Comment by zone117x** [comment]
> Should be reviewd with ignore whitespace. `.editorconfig` forces spaces for cs but tt files had tabs, so i converted to spaces. Should this be done in a separate pr to be merged before this?

Fine this PR :)

File: MimeMapping.csproj (hunk 3 above)
Comment (zone117x) [inline MimeMapping.csproj:7]: Can you update this repo's readme with a similar change?

Comment (viceice) [reply MimeMapping.csproj:7]: link is below in `See` section

**Comment by viceice** [comment]
- Updated readme and description to mostly match (description doesn't support markdown)
- I've now embedding the readme to the nuget package, so it's preferred by nuget.org to display[^1]
- Add sourcelink support for better consumer debugging experience[^2]


[^1]: https://devblogs.microsoft.com/nuget/add-a-readme-to-your-nuget-package/
[^2]: https://github.com/dotnet/sourcelink#source-link

**Comment by viceice** [comment]
here you can see the new readme in action: https://www.nuget.org/packages/VisualOn.MimeMapping/#readme-body-tab

**Review by zone117x — APPROVED** [review]
Thanks again!

**Pull request merged**